File: MoodByte/src/main/java/org/example/Modelo/Entrada.java
package org.example.Modelo;

import jakarta.persistence.*;

import java.time.LocalDate;

@Entity
@Table(
        name = "entradas",
        uniqueConstraints = {
                @UniqueConstraint(columnNames = {"idDiario", "fechaEntrada"})
        }
)
public class Entrada {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    @Column(columnDefinition = "TEXT")
    private String texto;
    private LocalDate fechaEntrada;
    @ManyToOne
    @JoinColumn(name="idDiario")
    private Diario diario;

    public Entrada() {}

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getTexto() {
        return texto;
    }

    public void setTexto(String texto) {
        this.texto = texto;
    }

    public LocalDate getFechaEntrada() {
        return fechaEntrada;
    }

    public void setFechaEntrada(LocalDate fechaEntrada) {
        this.fechaEntrada = fechaEntrada;
    }

    public Diario getDiario() {
        return diario;
    }

    public void setDiario(Diario diario) {
        this.diario = diario;
    }
}

File: MoodByte_App/app/src/main/java/com/example/moodbyte/domain/model/Entrada.kt
package com.example.moodbyte.domain.model

import com.example.moodbyte.data.remote.dtos.EntradaDto
import java.time.LocalDate

data class Entrada(
    val id:Long,
    val texto:String,
    val fechaEntrada: LocalDate,
    val idDiario: Long
    )

fun Entrada.toDto() = EntradaDto(
    id = id,
    texto = texto,
    fechaEntrada = fechaEntrada.toString(),
    idDiario = idDiario
)
